sdk/ochopod/models/piped.py

```diff
@@ -381,6 +391,19 @@ def on(self, data):
         self.commands.popleft()
         return 'spin', data, 0
 
+    def _pipe(self, proc):
+        #
+        # - Log any stdout or stderr from data.sub by polling the Popen in data.sub
+        #
+        while True:
+            nextline = proc.stdout.readline().rstrip('\n')
+            code = proc.poll()
+            if nextline == '' and code is not None:
+                logger.warning('(pid {0}): Terminated with code {1}.'.format(proc.pid, code))
```

Comment:
also i would simplify the logic a bit .. no need to warn if the process exited since the actor will do it anyway .. just break out of the loop if you can't stream anymore and that's it